bokehjs/src/coffee/models/renderers/glyph_renderer.coffee

```diff
@@ -230,7 +230,10 @@ export class GlyphRendererView extends RendererView
     @glyph.draw_legend_for_index(ctx, x0, x1, y0, y1, index)
 
   hit_test: (geometry) ->
-    @glyph.hit_test(geometry)
+    if @model.visible
+      return @glyph.hit_test(geometry)
+    else
+      return null
```

Comment:
I would suggest putting this in `Glyph.hit_test` instead, where it would be much easier to add a test for (and then suggest adding a unit test to make sure `null` is returned if `glyph.visible = false`)